packages/theming-core/src/generics/themeRegistry.ts

```diff
@@ -0,0 +1,247 @@
+/**
+ * Function which takes a parent theme, does some processing on it, and returns a new theme definition.
+ * This definition will then be passed to the standard resolver
+ *
+ * @internal This is an experimental interface and will be changed post design review.
+ */
+export type ProcessTheme<ITheme, IThemeDefinition> = (parentTheme: ITheme) => IThemeDefinition;
+
+/**
+ * Function which takes a theme definition and a parent theme and produces a new resolved theme
+ *
+ * @internal This is an experimental interface and will be changed post design review.
+ */
+export type ResolveTheme<ITheme, IThemeDefinition> = (definition: IThemeDefinition | undefined, parentTheme: ITheme) => ITheme;
```

Comment:
Could also consider:
```suggestion
export type ResolveTheme<ITheme, IThemeDefinition> = (parentTheme: ITheme, definition?: IThemeDefinition) => ITheme;
```
If structured this way, callers don't have to provide an explicit `undefined` argument for `definition`.

Reply:
Come to think of it, if `definition` is `undefined`, isn't the output just `parentTheme`? Should `definition` be required?